src/MetadataManager/pathToHash.ts

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as vscode from 'vscode';
+import * as crypto from 'crypto';
+import fs from 'fs';
+import {Metadata} from './metadataAPI';
+import {MetadataEventManager} from './EventManager';
+
+export class PathToHash{
+    private static instance: PathToHash;
+    private pathToHash: any;
+
+    private constructor() {
+    }
+    
+    public static async getInstance() {
+        if (!this.instance) {
+            this.instance = new PathToHash();
+            this.instance.pathToHash = await this.instance.initPathToHash();
+            await this.instance.validateMetadata(this.instance.pathToHash);
+        }
+        return this.instance;
+    }
+
+    private async initPathToHash() {
```

Comment:
```suggestion
    private async init() {
```

There is no need to repeat the classname 'PathToHash'